src/init.cpp

```diff
@@ -1781,5 +1784,19 @@ bool AppInitMain(InitInterfaces& interfaces)
         g_banman->DumpBanlist();
     }, DUMP_BANS_INTERVAL * 1000);
 
+    // Once initial sync is finished, flush state to protect against data loss
+    bool did_initial_flush = false;
+    int last_chain_height = -1;
+    scheduler.scheduleEvery([&did_initial_flush, &last_chain_height] {
```

Comment:
Lifetime issue with `did_initial_flush` and `last_chain_height` when `AppInitMain` goes out of scope?

Reply:
Moved them inside the lambda as static variables.